platform/keysystem/keyserver/keyapi/api.go

```diff
@@ -71,8 +77,39 @@ func (k *ConfiguredKeyserver) GetClientCAs() *x509.CertPool {
 	return k.Context.AuthenticationAuthority.ToCertPool()
 }
 
-func (k *ConfiguredKeyserver) GetServerCert() tls.Certificate {
-	return k.Context.ServerTLS.ToHTTPSCert()
+const RenewalMargin = time.Minute * 5
+const ValidityInterval = time.Hour * 24
+
+func (k *ConfiguredKeyserver) GetValidServerCert(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	k.CertLock.Lock()
+	defer k.CertLock.Unlock()
+
+	if k.ServerCert != nil && time.Now().Add(RenewalMargin).Before(k.ServerCert.Leaf.NotAfter) {
+		// we still have a valid certificate, so use that
+		k.Logger.Printf("Using existing server certificate valid until %v", k.ServerCert.Leaf.NotAfter)
```

Comment:
how often does this log?  if it's the normal case there's probably no need to do this

Reply:
Pretty often. You're right that I should remove this.